src/pybamm/config.py

```diff
@@ -69,44 +177,28 @@ def ask_user_opt_in(timeout=10):
         "For more information, see https://docs.pybamm.org/en/latest/source/user_guide/index.html#telemetry"
     )
 
-    def get_input():  # pragma: no cover
-        try:
-            user_input = (
-                input("Do you want to enable telemetry? (Y/n): ").strip().lower()
-            )
-            answer.append(user_input)
-        except Exception:
-            # Handle any input errors
-            pass
+    user_input = get_input_or_timeout(timeout)
 
-    time_start = time.time()
+    if user_input is None:
+        print("\nTimeout reached. Defaulting to not enabling telemetry.")
+        return False
 
     while True:
-        if time.time() - time_start > timeout:
-            print("\nTimeout reached. Defaulting to not enabling telemetry.")
+        if user_input == "":  # Empty input should mean a no
+            print("Telemetry enabled.\n")
             return False
-
-        answer = []
-        # Create and start input thread
-        input_thread = threading.Thread(target=get_input)
-        input_thread.daemon = True
-        input_thread.start()
-
-        # Wait for either timeout or input
-        input_thread.join(timeout)
-
-        if answer:
-            if answer[0] in ["yes", "y", ""]:
-                print("\nTelemetry enabled.\n")
-                return True
-            elif answer[0] in ["no", "n"]:
-                print("\nTelemetry disabled.\n")
-                return False
-            else:
-                print("\nInvalid input. Please enter 'yes/y' for yes or 'no/n' for no.")
-        else:
-            print("\nTimeout reached. Defaulting to not enabling telemetry.")
+        elif user_input.lower() in ["y", "yes"]:
+            print("Telemetry enabled.\n")
+            return True
+        elif user_input.lower() in ["n", "no"]:
+            print("Telemetry disabled.")
             return False
+        else:
+            print("Invalid input. Please enter 'Y/y' for yes or 'n/N' for no:")
+            user_input = get_input_or_timeout(timeout)
+            if user_input is None:
+                print("\nTimeout reached. Defaulting to not enabling telemetry.")
+                return False
```

Comment:
I know I brought it up before, but if we just check for the yes options, we can skip having an infinite loop

Reply:
If a person presses "Enter" in haste, that enables telemetry for them (because of your suggestion that I applied above), which might not be what they want – I'm trying to reduce the chance of that happening

Reply:
So, should I keep it as "yes" if Enter is pressed or "no"? I feel the latter is the better option